core/src/View/Kokkos_BasicView.hpp

```diff
@@ -96,6 +96,14 @@ transform_kokkos_slice_to_mdspan_slice(const T &s) {
   return KokkosSliceToMDSpanSliceImpl<T>::transform(s);
 }
 
+// Default implementation for computing allocation size (in #of elements)
+// from mapping and accessor.
```

Comment:
Since this customization is not used in Kokkos itself, a bit more explanation spelling out that this the default for types where `#of elements*sizeof(element) ==  allocation size` might be good. After all we introduce an interface using AccessorType that we don't use in the implementation.